kernel/src/syscall.rs

```diff
@@ -73,36 +80,63 @@ pub trait UserspaceKernelBoundary {
     /// initialization must happen in the `initialize_process()` function.
     type StoredState: Default;
 
+    /// Called by the kernel during process creation to inform the kernel of the
+    /// minimum amount of process-accessible RAM needed by a new process. This
+    /// allows for architecture-specific process layout decisions, such as stack
+    /// pointer initialization.
+    ///
+    /// This returns the minimum number of bytes of process-accessible memory
+    /// the kernel must allocate to a process so that a successful context
+    /// switch is possible.
+    ///
+    /// Some architectures may not need any allocated memory, and this should
+    /// return 0. In general, implementations should try to pre-allocate the
+    /// minimal amount of process-accessible memory (i.e. return as close to 0
+    /// as possible) to provide the most flexibility to the process. However,
+    /// the return value will be nonzero for architectures where values are
+    /// passed in memory between kernelspace and userspace during syscalls or a
+    /// stack needs to be setup.
+    fn initial_process_app_brk_size(&self) -> usize;
```

Comment:
I suspect there should be an alignment-related statement here. Something like "The memory will be aligned to at least the smallest alignment supported by the MPU."

This might be a bit pedantic, because I don't expect there to be an architecture where MPU regions are not at least word-aligned.

Reply:
If you are thinking about alignment for the MPU, process.rs is not responsible for aligning memory for the MPU. process.rs doesn't know what the MPU alignment requirements are just like it doesn't know the required min app_brk size, since both are platform-specific.

Yes, the platform will want the memory region of at least this sized aligned for the MPU, but it's up to the platform's MPU implementation to enforce that.

That being said, process.rs does decide where inside of the entire memory allocation for a process to allocate the number of bytes reserved from this function. I think it is established that the "process accessible" region is in the lower addresses of the process's overall memory region. I'm not sure if we explicitly say that the kernel will start the process accessible region from the very beginning of the overall region, however. Although it is certainly implied in this overall interface.

Reply:
> That being said, process.rs does decide where inside of the entire memory allocation for a process to allocate the number of bytes reserved from this function. I think it is established that the "process accessible" region is in the lower addresses of the process's overall memory region. I'm not sure if we explicitly say that the kernel will start the process accessible region from the very beginning of the overall region, however. Although it is certainly implied in this overall interface.

Ah, I misunderstood how that works. I suppose that means that `process.rs` is responsible for suitably aligning the region?

Reply:
process.rs can start the process's accessible memory region wherever it wants inside of the region chosen by the MPU code, so yes it does need to think about alignment. Should we say that accessible memory start will be word aligned?

Reply:
Alright I added it to the `initialize_process` comment. I made a comment in process.rs that we have to respect this as well.